roles/argocd_config/tasks/config-repo.yml

```diff
@@ -0,0 +1,80 @@
+---
+- name: Validate Repo without SSH key
+  ansible.builtin.assert:
+    that:
+      - ac_repo | regex_search('^http[s]?://')
+      - ac_gh_token is defined
+      - ac_gh_username is defined
+    fail_msg:
+      - "Repository URL must start with 'http://' or 'https://'."
+      - "GitHub token and username must be defined when SSH key is not provided."
+  when:
+    - ac_ssh_key is not defined
+
+- name: Validate Repo with SSH key
+  ansible.builtin.assert:
+    that:
+      - ac_repo | regex_search('^\w+@')
+      - ac_gh_token is undefined
```

Comment:
I see that token-based and ssh-key-based auth work on the most common SCM tools, can get do not tie this to GH?

ac_scm_token?
ac_scm_token?
ac_scm_username?